gammapy/background/models.py

```diff
@@ -650,3 +651,26 @@ def compute_rate(self):
         bg_rate = bg_rate.to('MeV-1 sr-1 s-1')
 
         self.bg_rate.data = bg_rate
+
+    def smooth(self):
+        """Smooth the bkg rate with a gaussian 1D kernel.
+
+        """
+        for binE, E in enumerate(self.counts.energy[:-1]):
+            counts = self.counts.data[binE, :]
+            Nev = np.sum(counts).value
+            if (Nev > 0):
+                Np = len(counts)
```

Comment:
I think the code would become easier to understand if you factor the part starting with `Np = len(counts)` down to the `convolve` call into a separate function `_poisson_gauss_smooth` (or something).
At the moment this method is quite complex, because you loop energy bands and then have an algorithm for each energy band and then put the result back in the 2D array. It would be better to factor the 1D algorithm out into a utility function.

So something like this (pseudo-code);
```
for idx in range(len(self.counts.energy)):
    counts = self.counts.data[idx]
    bkg = self.bkg.data[idx]
    bkg_smooth = _poisson_gauss_smooth(counts, bkg)
    self.bkg.data[idx] = bkg_smooth
```